### PR TITLE
abi: set default umask and rlimits

### PR DESCRIPTION
**Pull request by giuseppe**
the code got lost in the migration to podman 2.0, reintroduce it.

Closes: https://github.com/containers/podman/issues/6989

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
Fixes: https://github.com/containers/podman/issues/6989

**Comment by rhatdan** [comment]
@giuseppe good catch.

File: cmd/podman/root.go [hunk 1]
```diff
@@ -171,6 +171,13 @@ func persistentPreRunE(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	if !registry.IsRemote() {
+		err := registry.ContainerEngine().EarlyProcessSetup(registry.Context())
```
Comment (mheon) [inline cmd/podman/root.go:175]: This might be better as part of `cobra.OnInitialize()` - we're already configuring logging there

Comment (mheon) [reply cmd/podman/root.go:175]: (Most notably, removes the need for a public function to do the work)

Comment (giuseppe) [reply cmd/podman/root.go:175]: are we able there to distinguish between the remote and the ABI case?

Comment (giuseppe) [reply cmd/podman/root.go:175]: well probably it makes no sense to distinguish, we can run this in both cases

File: cmd/podman/early_init_linux.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	"fmt"
+	"os"
+	"syscall"
+
+	"github.com/pkg/errors"
+)
+
+func setRLimits() error {
+	rlimits := new(syscall.Rlimit)
+	rlimits.Cur = 1048576
```
Comment (vrothberg) [inline cmd/podman/early_init_linux.go:13]: Nit: Can we make this a constant in `define`? It's also being used in `pkg/spec`.

Comment (giuseppe) [reply cmd/podman/early_init_linux.go:13]: yes that is a good idea.  Do we know where we took this magic number and what it means?

Comment (giuseppe) [reply cmd/podman/early_init_linux.go]: called `define.RLimitDefaultValue` for now

**Pull request merged**